src/components/TodoList/TodoList.tsx

```diff
@@ -1,224 +1,123 @@
 /* eslint-disable */
+import { useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { getTodos } from '../../api';
+import { todosSlice } from '../../features/todos';
+import cn from 'classnames';
+import { currentTodoSlice } from '../../features/currentTodo';
 import React from 'react';
+import { Status } from '../../types/Status';
+import { Todo } from '../../types/Todo';
+
+function filterTodos(todos: Todo[], query: string, status: Status) {
+  let filteredTodos = todos;
+  if (query) {
+    filteredTodos = todos.filter((todo: any) =>
+      todo.title.toLowerCase().includes(query.toLowerCase()),
+    );
+  }
+  if (status !== Status.All) {
+    filteredTodos = filteredTodos.filter(
+      (todo: any) => todo.completed === (status === 'completed'),
+    );
+  }
+  return filteredTodos;
+}
 
 export const TodoList: React.FC = () => {
+  const { fetchTodos } = todosSlice.actions;
+  const { setCurrentTodo } = currentTodoSlice.actions;
+  const dispatch = useDispatch();
+  const todos = useSelector((state: any) => state.todosReducer);
+  const currentTodo = useSelector((state: any) => state.currentTodoReducer);
+  const { query, status } = useSelector((state: any) => state.filterReducer);
+  let visibleTodos = filterTodos(todos, query, status);
+
+  useEffect(() => {
+    getTodos().then(resp => {
+      dispatch(fetchTodos(resp));
+    });
+  }, []);
+
+  const handleSetCurrentTodo = (todo: any) => {
```

Comment:
Don't use `any` as type, fix it everywhere
```suggestion
  const handleSetCurrentTodo = (todo: Todo) => {
```